src/main/java/com/arpnetworking/metrics/common/sources/CollectdHttpSourceV1.java

```diff
@@ -0,0 +1,290 @@
+                        .map(ByteString::toArray) // Transform to array form
+                        .map(this::parseRecords) // Parse the json string into a record builder
+                                                 // NOTE: this should be _parser::parse, but aspectj NPEs with that currently
+                        .named("parseJson");
+
+                final Flow<HttpRequest, Map<String, String>, NotUsed> parseHeadersFlow = Flow.<HttpRequest>create()
+                        .map(HttpRequest::getHeaders)
+                        .map(Actor::extractTagHeaders)
+                        .named("parseHeaders");
+
+                final Flow<
+                        Pair<Map<String, String>, List<DefaultRecord.Builder>>,
+                        DefaultRecord.Builder,
+                        NotUsed> applyTagsFlow = Flow.<Pair<Map<String, String>, List<DefaultRecord.Builder>>>create()
+                                .flatMapConcat(Actor::mapExpandPair)
+                                .map(Actor::applyTags)
+                                .named("applyTags");
+
+                final Flow<DefaultRecord.Builder, Record, NotUsed> buildFlow = Flow.<DefaultRecord.Builder>create()
+                        .map(DefaultRecord.Builder::build);
+
+                // Shapes
+                final UniformFanOutShape<HttpRequest, HttpRequest> split = builder.add(Broadcast.create(2));
+                final FlowShape<HttpRequest, List<DefaultRecord.Builder>> parseJson = builder.add(parseJsonFlow);
+                final FlowShape<HttpRequest, Map<String, String>> parseHeaders = builder.add(parseHeadersFlow);
+                final FanInShape2<
+                        Map<String, String>,
+                        List<DefaultRecord.Builder>,
+                        Pair<Map<String, String>, List<DefaultRecord.Builder>>> join = builder.add(Zip.create());
+                final FlowShape<
+                        Pair<Map<String, String>, List<DefaultRecord.Builder>>,
+                        DefaultRecord.Builder> applyTags = builder.add(applyTagsFlow);
+                final FlowShape<DefaultRecord.Builder, Record> build = builder.add(buildFlow);
+
+                // Wire the shapes
+                builder.from(split.out(0)).via(parseHeaders).toInlet(join.in0());
+                builder.from(split.out(1)).via(parseJson).toInlet(join.in1());
+                builder.from(join.out()).via(applyTags).via(build);
+
+                return new FlowShape<>(split.in(), build.out());
+            });
+        }
+
+        private static DefaultRecord.Builder applyTags(final Pair<Map<String, String>, DefaultRecord.Builder> pair)
+                throws Exception {
+            final DefaultRecord.Builder recordBuilder = pair.second();
+            final Map<String, String> annotations = pair.first();
+            recordBuilder.setAnnotations(annotations);
+            applyTag("service", annotations, recordBuilder::setService);
```

Comment:
I believe we filter these special annotations out of the annotations map into the specific fields in the file parsing path.